src/main/kotlin/br/com/gmfonseca/coding/Main.kt

```diff
@@ -1,5 +1,23 @@
 package br.com.gmfonseca.coding
 
+const val isStuff = false
+
+data class Something(val isActive: Boolean = true)
+
 fun main() {
+    foobar(true, Something(false))
+}
 
-}
\ No newline at end of file
+fun foobar(isSome: Boolean, something: Something? = null): String {
+    val isSomethingActive = something != null && something.isActive
+    // ...
+    if (something == null || isSomethingActive) {
```

Comment:
Ah, essa condicao ta redundante, poderia reduzir ela para algo como:
```suggestion
    val isSomethingActive = something == null || something.isActive

    if (isSomethingActive) {
```